Sources/App/Core/Search.swift

```diff
@@ -201,25 +224,27 @@ enum Search {
             fatalError("Database must be an SQLDatabase ('as? SQLDatabase' must succeed)")
         }
 
-        let sanitizedTerms = sanitize(terms)
-        guard !sanitizedTerms.isEmpty else {
+        let unfilteredSanitizedTerms = sanitize(terms)
+        guard !unfilteredSanitizedTerms.isEmpty else {
             return nil
         }
+        
+        let (sanitizedTerms, filters) = SearchFilterParser().separate(terms: unfilteredSanitizedTerms)
 
         // page is one-based, clamp it to ensure we get a >=0 offset
         let page = page.clamped(to: 1...)
         let offset = (page - 1) * pageSize
         let limit = pageSize + 1  // fetch one more so we can determine `hasMoreResults`
 
-        // only include non-package results on first page
-        let query = (page == 1)
+        // only include non-package results on first page and only if there are no filters applied
+        let query = (page == 1 && filters.isEmpty)
```

Comment:
This is the only thing that is stopping keyword/author results from showing in search results when filters are applied (the brief discussion which happened on Discord).